Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -211,6 +211,54 @@ bool TextureCache::TCacheEntryBase::OverlapsMemoryRange(u32 range_address, u32 r
 	return true;
 }
 
+void TextureCache::TCacheEntryBase::DoPartialTextureUpdates()
+{
+	const bool isPaletteTexture = (format== GX_TF_C4 || format == GX_TF_C8 || format == GX_TF_C14X2 || format >= 0x10000);
+
+	// As long as there's no upscaling support, there's no point in doing this when doing efb to ram
+	// TODO: Only update normal textures. Would there be any point in updating efb copies?
+	// TODO: Are texture updates for paletted textures possible?
+	// Efb copies might be scaled, while normal textures are not. So don't try to update normal textures with scaled efb copies
+	if (!g_ActiveConfig.backend_info.bSupportsCopySubImage || !g_ActiveConfig.bSkipEFBCopyToRam || IsEfbCopy() || isPaletteTexture
+		|| (g_ActiveConfig.bCopyEFBScaled && g_ActiveConfig.iEFBScale != SCALE_1X))
+		return;
+
+	u32 block_width = TexDecoder_GetBlockWidthInTexels(format);
+	u32 block_height = TexDecoder_GetBlockHeightInTexels(format);
+	u32 block_size = block_width * block_height * TexDecoder_GetTexelSizeInNibbles(format) / 2;
+
+	u32 expanded_width = (native_width + block_width - 1) & ~(block_width - 1);
+
+	u32 numBlocksX = expanded_width / block_width;
```

Comment:
u32 numBlocksX = (native_width + block_width - 1) / block_width;

There is no need to also calculate expanded_width.